common/primitives/core/src/omni/intent.rs

```diff
@@ -52,10 +50,14 @@ pub struct TransferSolana {
     pub value: u64,
 }
 
-#[derive(Encode, Decode, Debug, Clone, PartialEq, Eq, MaxEncodedLen, TypeInfo)]
+#[derive(Encode, Decode, Debug, Clone, PartialEq, Eq, TypeInfo)]
 pub struct Asset {
-    pub id: AssetId,
-    pub chain_id: ChainId,
+    // `name` is intentionally defined as string-like type to:
+    // - allow easier integration with external service. E.g. APIs of binance/pumpx definitely expect strings
+    // - be future-compatible: meme-coins/alt-coins could have arbitrary names
+    pub name: Vec<u8>,
```

Comment:
This basically doesn't align well with [current token amount query implementation](https://github.com/litentry/heima/blob/dev/tee-worker/omni-executor/intent/token-query/src/lib.rs#L31-L41).

Should we query pumpx for account balance ?

Reply:
As per requirements, we would only swap "mainstream coins" (= native currency and stable coins) for any coin, specifically on solana it would be [`SOL` and `USDT`](https://linear.app/litentry/issue/P-1392/double-check-supported-sourcedest-chains-token-types#comment-9513941f). So I assume we only need to query balance of those two tokens.

Shall we have a `to_solana_token()` which maps the `name` on Solana to `SolanaToken` type? We'll need to hardcode the mint address of `usdt` in the mapping (https://solscan.io/token/Es9vMFrzaCERmJfrF4H2FYD4KCoNkY11McCe8BenwNYB)

Reply:
I think mint token / erc20 address should be used instead of `name` in `intent.rs` - this is what uniiquely identiefies token.
We can implement whitelisting if limitation is needed.

If pumpx api cannot accept address and we have to provide name - we should do the mapping in package responsible for integrating with pumpx.

Reply:
pumpx api expects CA https://github.com/litentry/heima/pull/3303/files#diff-649b162768ba7355b16233723b2ffa0ab2606dc5a87fe55bee1692992219d29bR66

Reply:
CA is string tho, shall we also use Vec<u8> for it? Especially considering different chains could have different formats.

And what about binance-api? Does it expect token name or CAs? @silva-fj

Reply:
The Binance API expects token pair symbols e.g: USDTSOL to trade USDT -> SOL

Reply:
Hmm then we'll need `name` <> `CA` mapping in the code anyway. In this case I prefer we simply use `name` for better readability, not to mention `CA` has to be of type `Vec<u8>` too even if go for it

Reply:
Wdyt? @kziemianek @silva-fj

Reply:
But it can be enum `Asset { Solana(SolanaAsset) , Ethereum(EthereumAsset) }` , see asset enums [here](https://github.com/litentry/heima/blob/dev/tee-worker/omni-executor/intent/token-query/src/lib.rs#L31-L41).

Additionaly we will need tickers for only few assets (USDT,SOL,ETH) while we will need `CA` for all of the rest (practically unbounded). If that's correct I would prefer to map address to ticker.

Reply:
So an enum whose entries are grouped by chain types, sounds like a good idea! Let me adjust it